src/jit/importer.cpp

```diff
@@ -3805,6 +3805,14 @@ GenTree* Compiler::impIntrinsic(GenTree*                newobjThis,
             JITDUMP("\nimpIntrinsic: Expanding %sSpan<T>.get_Item, T=%s, sizeof(T)=%u\n", isReadOnly ? "ReadOnly" : "",
                     info.compCompHnd->getClassName(spanElemHnd), elemSize);
 
+            if (varTypeIsStruct(impStackTop(0).val))
```

Comment:
The VM should not be returning `CORINFO_INTRINSIC_ReadOnlySpan_GetItem` for indexer that returns struct today. Sounds like something else is off...